peer.go

```diff
@@ -191,26 +134,11 @@ func IDFromPublicKey(pk ic.PubKey) (ID, error) {
 	if err != nil {
 		return "", err
 	}
-	hash, _ := mh.Sum(b, mh.SHA2_256, -1)
-	return ID(hash), nil
-}
-
-// IDFromEd25519PublicKey returns the Peer ID corresponding to Id25519 pk
-func IDFromEd25519PublicKey(pk ic.PubKey) (ID, error) {
-	b, err := pk.Bytes()
-	if err != nil {
-		return "", err
+	var alg uint64 = mh.SHA2_256
+	if len(b) <= 40 {
```

Comment:
you might want to define the algorithm and the magic 40 as consts.

Reply:
And now the world is a little less magical :crying_cat_face:.